fdbclient/MonitorLeader.actor.cpp

```diff
@@ -968,11 +989,16 @@ ACTOR Future<MonitorLeaderInfo> monitorProxiesOneGeneration(
 			shrinkProxyList(ni, lastCommitProxyUIDs, lastCommitProxies, lastGrvProxyUIDs, lastGrvProxies);
 			clientInfo->setUnconditional(ni);
 			successIndex = index;
+			allConnectionsFailed = false;
 		} else {
 			TEST(rep.getError().code() == error_code_failed_to_progress); // Coordinator cant talk to cluster controller
 			TEST(rep.getError().code() == error_code_lookup_failed); // Coordinator hostname resolving failure
+			TraceEvent("MonitorProxiesConnectFailed")
+			    .detail("Error", rep.getError().name())
+			    .detail("Coordinator", clientLeaderServer.getAddressString());
 			index = (index + 1) % coordinatorsSize;
-			if (index == successIndex) {
+			allConnectionsFailed = (index == successIndex);
```

Comment:
I think this should be changed to
`if (index == successIndex) allConnectionsFailed = true;`
Otherwise we can unintentionally set it back to false?

Reply:
I think this technically might work as is because it will still check for updates to the cluster file once per round that it goes through the coordinators unsuccessfully. It's probably still clearer and a little more efficient to write it as you've suggested, though.

Reply:
Good point. I changed it as suggested

Reply:
Actually, I'm questioning myself now. If we write this way, when we fail to connect all coordinators, and start another round, we will persist `every` time before we successfully connect to one. So maybe what you wrote at first makes more sense? Or
```
if (index == successIndex) {
    allConnectionsFailed = true;
    wait(delay(CLIENT_KNOBS->COORDINATOR_RECONNECTION_DELAY));
} else {
    allConnectionsFailed = false;
}
```

Reply:
I think it is ok to keep the flag true as originally suggested by you. The flag is just an indication that we should try to update the connection string from the cluster file if it has changed. If the file does not change, nothing happens. And if the file changes, the connection string is updated, the actor returns and a new monitoring loop is started.